sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1101,7 +1101,7 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
    * Create a [[CreateStruct]] expression.
    */
   override def visitRowConstructor(ctx: RowConstructorContext): Expression = withOrigin(ctx) {
-    CreateStruct(ctx.expression.asScala.map(expression))
+    CreateStruct(ctx.expression.asScala.map(expression)).toCreateNamedStruct
```

Comment:
@cloud-fan this process tow-constructor expression of the form `(1,"a") as col1(a,b)`.
I could basically leave this as a `CreateStruct` but than I'd had to do something like transformDown in `visitInlineTable` which is basically the reason for this mess.